### ISSUE TITLE
Improve Rule Search

### ISSUE THREAD

**Issue by Jay-Karia** [eslint/eslint#18989]
I have a suggestion to add tags/labels to all the rules for ease of searching and filtering them, as users can easily get their required rules.

###### Adding labels to rule
Add a new property `labels` to the rules in the `meta` object
```js
module.exports = {
    meta: {
        // ...
        labels: ["label_1", "label_2"]
    },
    create: function(context) {
        // ...
    }
};
```

###### How users can use this feature
User can get the filtered rules from the filter above the rules where users can select one or more labels.
The UI would consists of "chips" for the labels OR a dropdown menu.

![eslintFilter](https://github.com/user-attachments/assets/f83c7b66-3e5a-4792-9b46-bb1c056b5283)
Adding the filter here

###### Implementation

Create an array of all the labels (say `allLabels`) and expose it to the documentation which would help to render the filter component with the labels in it. Can modify the labels from the array.
By doing this we have to ensure that the `labels` array passed to the rule should contain the listed labels in `allLabels`.

This is the simplest implementation I can thought for as I do not want to make it complicated and time consuming.
I would like to hear your thoughts on it and I am ready to modify it if required,

###### Examples

- `no-constant-condition` - ["variable", "condition", "ternary"]
- `no-dupe-args` - ["variable", "function"]
- `no-unused-vars` - ["variable"]
- `no-else-return` - ["if-else", "condition"]

A rough draft.
Not thought about the labels for all the rules.

**Comment by Jay-Karia** [comment]
@Tanujkanti4441 
Yeah, but what if I do not know the exact name of the rule and just wanted to know about the rules on specific category, and would add the required ones.

Let's say I want rules related to "conditions" , then I would filter all the rules under that category and would add them.

This would significantly reduce the chances of missing out important or crucial rules.

**Comment by Tanujkanti4441** [comment]
Search works on keyword to find pages as:
![Image](https://github.com/user-attachments/assets/90bee416-dbc2-480c-a866-44e1ad16c888)

It Even matches the content of the pages
![Image](https://github.com/user-attachments/assets/c1669e63-3b5c-49b4-9678-ca7b1fc533d0)
